doc/source/dev/contributor/debugging_linalg_issues.rst

```diff
@@ -0,0 +1,440 @@
+    YMMV). In case you do want to try:
+
+    Build reference BLAS and LAPACK:
+
+        $ git clone https://github.com/Reference-LAPACK/lapack.git
+        $ cd lapack
+        $ mkdir build && cd build
+        $ cmake -DCBLAS=ON -DBUILD_SHARED_LIBS=OFF ..
+        $ cmake --build . -j
+        $ cmake --install . --prefix $PWD/../../built-libs
+
+    Then add the following two lines to the ``cmake ..`` configure command for
+    FlexiBLAS::
+
+        -DSYS_BLAS_LIBRARY=$PWD/../../built-libs/lib/libblas.a \
+        -DSYS_LAPACK_LIBRARY=$PWD/../../built-libs/lib/liblapack.a \
+
+
+Creating reproducers in C or Fortran
+------------------------------------
+
+Our experience tells us that a large majority of bugs are inside SciPy rather
+than in OpenBLAS or another BLAS library. If the testing with different BLAS
+libraries tells us though that the problem is specific to a single BLAS library
+(maybe even a single version of that library with a regression), the next step
+is to produce a reproducer in C or Fortran; doing so is necessary for reporting
+the bug upstream, and makes it much easier for the BLAS library developers to
+address the problem.
+
+To get from a Python reproducer which uses a ``scipy`` function with NumPy
+arrays as input to a C/Fortran reproducer, it is necessary to find the code
+path taken in SciPy and determine which exact BLAS or LAPACK function is
+called, and with what inputs. This can then be reproduced in C/Fortran
+by defining some integer/float variables and arrays (typically small arrays
+with hardcoded numbers are enough).
+
+Below a reproducer is shown for an issue in reference LAPACK, which was
+reported as a SciPy issue in `scipy#11577
+<https://github.com/scipy/scipy/issues/11577>`__. We'll name the file
+``ggev_repro_gh_11577.c|f90``:
+
+.. tab-set::
+
+  .. tab-item:: C
+    :sync: C
+
+    .. literalinclude:: _code_examples/ggev_repro_gh_11577.c
+      :language: C
+
+
+  .. tab-item:: Fortran
+    :sync: Fortran
+
+    .. literalinclude:: _code_examples/ggev_repro_gh_11577.f90
+      :language: fortran
+
+Now we need to compile this reproducer locally and run it. If we're invoking a
+compiler directly, we need to add the needed compile and link flags by hand.
+The include path will depend on your local install, and the link flags will
+depend on which library you're testing. For example, to test against a local
+build of OpenBLAS:
+
+.. tab-set::
+
+  .. tab-item:: C
+    :sync: C
+
+    ::
+
+        $ gcc ggev_repro_gh_11577.c \
+          -I$PWD/../flexiblas-setup/built-libs/include/ \
+          -L$PWD/../flexiblas-setup/built-libs/lib -lopenblas
+        $ ./a.out  # to run the reproducer
+
+    For reference BLAS/LAPACK, the ``-lopenblas`` should be replaced with
+    ``-lblas -llapack``.
+
+  .. tab-item:: Fortran
+    :sync: Fortran
+
+    ::
+
+        $ gfortran ggev_repro_gh_11577.f90 \
+          -I/$PWD/../flexiblas-setup/built-libs/include/ \
```

Comment:
-I not needed locally, at least for a build in a conda env which only has a single lapack (tried netlib and openblas)

Reply:
Indeed, not needed there because the headers are in the default include dir; it is needed for the self-built libraries. I'll clarify.

Reply:
Added a long-ish note and repeated the commands without explicit paths.